src/mmw/apps/modeling/geoprocessing.py

```diff
@@ -52,6 +52,69 @@
 }
 
 
+@statsd.timer(__name__ + '.sjs_submit')
+def sjs_submit(host, port, args, data):
+    """
+    Submits a job to Spark Job Server. Returns its Job ID, which
+    can be used with sjs_retrieve to get the final result.
+    """
+    url = 'http://{}:{}/jobs?{}'.format(host, port, args)
+    response = requests.post(url, data=json.dumps(data))
+
+    if response.ok:
+        job = response.json()
+    else:
+        raise Exception('Unable to submit job to Spark JobServer.\n'
+                        'Details = {}'.format(response.text))
+
+    if job['status'] == 'STARTED':
+        return job['result']['jobId']
+    else:
+        raise Exception('Submitted job did not start in Spark JobServer.\n'
+                        'Details = {}'.format(response.text))
+
+
+@statsd.timer(__name__ + '.sjs_retrieve')
+def sjs_retrieve(host, port, job_id, retry=None):
+    """
+    Given a job ID, will try to retrieve its value. If the job is
+    still running, will call the optional retry function before
+    proceeding.
+    """
+    url = 'http://{}:{}/jobs/{}'.format(host, port, job_id)
+    response = requests.get(url)
+
+    if response.ok:
+        job = response.json()
+    else:
+        raise Exception('Unable to retrieve job {} from Spark JobServer.\n'
+                        'Details = {}'.format(job_id, response.text))
+
+    if job['status'] == 'FINISHED':
+        return job['result']
+    elif job['status'] == 'RUNNING':
+        if retry:
+            try:
+                retry()
+            except MaxRetriesExceededError, X:
+                delete = requests.delete(url)  # Job took too long, terminate
+                if delete.ok:
+                    raise X
```

Comment:
Maybe something like this would work:

```py
raise Exception('Job {} timed out, deleted.\n'
                'Details: {}'.format(job_id, delete.text))